asf_core_data/pipeline/preprocessing/feature_engineering.py

```diff
@@ -338,14 +342,32 @@ def get_heating_features(df, fine_grained_HP_types=False):
         (df["HP_INSTALLED"])
         | (df["SECONDHEAT_DESCRIPTION"].str.lower().str.contains("heat pump"))
         | (df["MAINHEAT_DESCRIPTION"].str.lower().str.contains("pumpa teas"))
+        | (df["SECONDHEAT_DESCRIPTION"].str.lower().str.contains("pumpa teas"))
         | (df["MAINHEAT_DESCRIPTION"].str.lower().str.contains("pwmp gwres"))
+        | (df["SECONDHEAT_DESCRIPTION"].str.lower().str.contains("pwmp gwres"))
         | (df["MAINHEAT_DESCRIPTION"].str.lower().str.contains("bwmp gwres"))
+        | (df["SECONDHEAT_DESCRIPTION"].str.lower().str.contains("bwmp gwres"))
         | (df["MAINHEAT_DESCRIPTION"].str.lower().str.contains("pympiau gwres"))
-        | (df["MAINHEAT_DESCRIPTION"].str.lower().str.contains("bympiau gwres")),
+        | (df["SECONDHEAT_DESCRIPTION"].str.lower().str.contains("pympiau gwres"))
+        | (df["MAINHEAT_DESCRIPTION"].str.lower().str.contains("bympiau gwres"))
+        | (df["SECONDHEAT_DESCRIPTION"].str.lower().str.contains("bympiau gwres")),
         True,
         False,
     )
 
+    # Update HP_TYPE, HEATING_SYSTEM and HEATING_FUEL after the changes to HP_INSTALLED
+    df["HEATING_SYSTEM"] = np.where(
+        df["HP_INSTALLED"] & (df["HP_TYPE"] == "No HP"),
+        "heat pump",
+        df["HEATING_SYSTEM"],
+    )
+    df["HEATING_FUEL"] = np.where(
+        df["HP_INSTALLED"] & (df["HP_TYPE"] == "No HP"), "electric", df["HEATING_FUEL"]
+    )
+    df["HP_TYPE"] = np.where(
+        df["HP_INSTALLED"] & (df["HP_TYPE"] == "No HP"), "heat pump", df["HP_TYPE"]
+    )
+
```

Comment:
Because we update `df["HP_INSTALLED"]` above, we also need to update the `HEATING_SYSTEM`, `HEATING_FUEL` AND `HP_TYPE` (`HP_TYPE` needs to be the last for it to work)